### ISSUE TITLE
Developer Guide: Document use of config.conf for plugins

### ISSUE THREAD

**Issue by jcsteh** [nvaccess/nvda#6050]
It makes sense for many plugins to use NVDA's central configuration system (`config.conf`) to store configuration. Among other things, this means settings can benefit from configuration profiles without any additional work. We should document this in the Developer Guide, including:

1. How to create a custom section;
2. How to read/write from that section; and
3. How (and why) to set a config spec for that section.

**Comment by nvdaes** [comment]
Hi, one question is if ConfigObj(StringIO) should be used, like in NVDA config, or if it would be better to avoid this, as done in audioscreen add-on.
Imo in audioscreen this is not used since the values are set "dinamically":
I am using this code in clipContentsDesigner add-on:

from configobj import ConfigObj
from cStringIO import StringIO
confspec = ConfigObj(StringIO("""
separator = string(default="")
"""))
config.conf.spec["clipContentsDesigner"] = confspec

Thanks.

**Comment by jcsteh** [comment]
I don't think there's a "better" answer here. If you only have a few 
settings, I'd probably recommend a dict, since it's more efficient than 
parsing from a string. However, if there are lots of settings, 
readability and maintainability with a dict might get ugly. In your 
case, I think you'd be better off with a dict, since you only have one 
setting, so there's actually more boilerplate using StringIO, etc. than 
if you'd just used a dict.

**Comment by nvdaes** [comment]
Thanks, your answer is very useful and I will implement it.

**Comment by nvdaes** [comment]
I'm using just:
confspec = {"separator": "string(default="")"}
Thanks.

**Comment by bdorer** [comment]
@Qchristensen @josephsl as you improoving documentation from time to time may you are able to do that? As we get more and more addons I wote for prioritizing this ticket

**Comment by derekriemer** [comment]
This prevents a lot of decentralization from occurring in the addons community, and has the possibility to streamline a lot of complex addons, which currently are used as reference code. This should be a fairly high priority documentation fix for developers.

**Comment by Adriani90** [comment]
@josephsl is the documentation of (config.conf) in process? I think at least in the addons dev guide it is somehow documented already. Right?

**Comment by josephsl** [comment]
Hi, at this point, I’d rather ask Leonard for this one. Thanks.

**Comment by Adriani90** [comment]
cc: @leonardder

**Comment by leonardder** [comment]
I'm not very efficient in devoting time to documentation writing I"m afraid. May be someone on the NVDA development list could start this project?

**Comment by josephsl** [comment]
Hi, at this point, Quentin and I are sort of documentation specialists… Once I get better (in terms of health), I’ll ask the community as to things that should be documented in the dev guide. Thanks.

**Comment by Adriani90** [comment]
cc: @Qchristensen

**Comment by Adriani90** [comment]
cc: @feerrenrut since the developer guide is maintained by NV Access, it would be good if you guys could take a look at this.

**Comment by feerrenrut** [comment]
Happy to take a PR to improve this, I'm sure there is lots of expertise in the add-on developer community.

**Comment by Adriani90** [comment]
I would add here some additional people from the addons community, it seems documenting this can become somehow complex. And I think a solid contribution would make this sustainable.
cc: @codeofdusk, @francipvb, @mltony, @abdel792, @Andre9642

**Comment by feerrenrut** [comment]
Suggestion: start small. It's much easier to get a PR accepted, especially for something as subjective as good documentation.

**Comment by Adriani90** [comment]
cc: @seanbudd, @SaschaCowley for triaging.
